File: scala/src/main/java/com/atguigu/java/test/Test1.java
package com.atguigu.java.test;

/**
 * @Description
 * @Author kaki
 * @create 2020-05-28-6:28 PM
 */
public class Test1 {

    public static void main(String[] args) {


        try {
            int a = 10;
            int b = 0;
            int c = a / b;
        } catch (ArithmeticException e){
            // catch时，需要将范围小的写到前面
            throw new ArithmeticException("NoGirlFriendException");
        } catch (Exception e){
            e.printStackTrace();
        } finally {
            System.out.println("finally");
        }


    }

}

File: scala/src/main/scala/com/atguigu/scala/test/Test1.scala
package com.atguigu.scala.test

import scala.collection.mutable

/**
  * @Description
  * @Author kaki
  * @create 2020-05-28-7:12 PM
  */
object Test1 {

  def main(args: Array[String]): Unit = {

    val arrays: Array[Int] = Array(30, 1, 34, 21, 89)

    val arr : Array[Int] = map(arrays, x => x * x)

    println(arr.toList)

  }

  def map(array: Array[Int],op: Int=>Int)  = {
    for (ele <- array) yield op(ele)
  }

}
